src/backend/distributed/connection/placement_connection.c

```diff
@@ -240,7 +240,7 @@ StartPlacementConnection(uint32 flags, ShardPlacement *placement, const char *us
 		userName = freeUserName = CurrentUserName();
 	}
 
-	key.placementId = placement->placementId;
+	key.placementId = placement->placement.placementId;
```

Comment:
We're introducing a lot of ambiguous naming in this last commit. I'm kind of inclined to omit the commit for the moment, merge this PR, and then maybe do the embedding a little more structurally (e.g. rename ShardPlacement to NodeShardPlacement and rename variables accordingly). What do you think?